nuxeo-core/nuxeo-core-management/src/main/java/org/nuxeo/ecm/core/management/probes/ProbeManagerImpl.java

```
@@ -218,4 +231,61 @@ protected void doRunProbe(ProbeInfo probe) {
         }
     }
 
+    @Override
+    public void registerProbeForHealthCheck(HealthCheckProbesDescriptor descriptor) {
+        String name = descriptor.getName();
+        if (!descriptor.isEnabled()) {
+            if (probesForHealthCheck.containsKey(name)) {
+                probesForHealthCheck.remove(name);
+                return;
+            }
+        }
+        if (infosByShortcuts.containsKey(name)) {
+            probesForHealthCheck.put(name, getProbeInfo(name));
+        }
+    }
+
+    @Override
+    public Collection<ProbeInfo> getHealthCheckProbes() {
+        return Collections.unmodifiableCollection(probesForHealthCheck.values());
+    }
+
+    @Override
+    public HealthCheckResult getOrRunHealthChecks() {
+        int refreshSeconds = Integer.parseInt(Framework.getProperty(DEFAULT_HEALTH_CHECK_INTERVAL_SECONDS_PROPERTY,
+                DEFAULT_HEALTH_CHECK_INTERVAL_SECONDS));
+        for (Entry<String, ProbeInfo> es : probesForHealthCheck.entrySet()) {
+            String probeName = es.getKey();
+            ProbeInfo probe = es.getValue();
+            if (probe == null) {
+                log.warn("Probe:" + probeName + " does not exist, skipping it for the health check");
+                continue;
+            }
+            getStatusOrRunProbe(probe, refreshSeconds);
+        }
+        return new HealthCheckResult(probesForHealthCheck.values());
+    }
+
+    @Override
+    public HealthCheckResult getOrRunHealthCheck(String name) {
+        int refreshSeconds = Integer.parseInt(Framework.getProperty(DEFAULT_HEALTH_CHECK_INTERVAL_SECONDS_PROPERTY,
+                DEFAULT_HEALTH_CHECK_INTERVAL_SECONDS));
+        if (!probesForHealthCheck.containsKey(name)) {
+            log.warn("Probe:" + name + " does not exist, or not registed for the healthCheck");
+            return new HealthCheckResult(Collections.emptyList());
```

Should throw a checked exception to be caught later by the servlet